### DISCUSSION TITLE
Consoles support?

### DISCUSSION THREAD

**Discussion by srodrigo** [rust-windowing/winit#2660] [General]
Hi :)

Where are we at regarding consoles support? I'm specially interested in Nintendo Switch. Is this in the roadmap?

Thanks.

**Comment by madsmtm** [comment]
There has been a bit of work on XBox here: https://github.com/rust-windowing/winit/issues/2506

And a general refactor of console-like events (I think) in https://github.com/rust-windowing/winit/issues/944.

Nintendo support has not really been discussed, it is quite niche so mostly there just hasn't been the drive to implement it (also, what is the status on `rustc` support on that currently? I think it has only actually received `core` support this year, so the ecosystem is probably also pretty young).

**Reply by srodrigo** [reply]
Thanks! I've got no idea about the status of rustc support. I'd probably say it's non existent.. But I don't have a dev kit.

**Comment by kchibisov** [comment]
>Where are we at regarding consoles support? I'm specially interested in Nintendo Switch. Is this in the roadmap?

I think you must use a proprietary toolkit to write stuff for it, so it's not really possible? Maybe the same for sony stuff. 

For Xbox it should be more or less possible.

**Reply by srodrigo** [reply]
> think you must use a proprietary toolkit to write stuff for it, so it's not really possible? Maybe the same for sony stuff.

Yeah, I think that's the issue. AFAIK the closest has been SDL2, which some people managed to make work on consoles. But SDL2 is C plus Rust bindings, not quite native Rust...

> For Xbox it should be more or less possible.

Interesting. I'm deciding between winit and SDL2. While wide portability is not currently a concern, It'd definitely be a nice to have. I know of some other folks who had to go with SDL2 because of this.

**Reply by heavyrain266** [reply]
SDL2 have official _fork_ which you request on dev forums of PS4/PS5 and Switch.

**Reply by notgull** [reply]
Makes me wonder if we should have an optionally-enabled backend that uses libSDL2, in order to be compatible in these edge cases...

**Reply by heavyrain266** [reply]
Why not just use pure SDL2 at this point? Pretty sure that it has some specific symbols on consoles so you cannot easly swap shared lib and still will have to modify winit's internal code.

**Reply by kchibisov** [reply]
Yeah, using pure SDL2 is fine. Proprietary consoles are out of scope for us, especially given that you need special compilers for them most of the time, which are provided by the manufacturer.
